### PR TITLE
Support POST form params as hashmaps

### PR DESCRIPTION
**Pull request by Amwam**

### CODE REVIEW

**Comment by mrak** [comment]
Thank you for the PR!

Instead of overloading `post`, we should add a `form:` option in a similar vein to #54. This would reduce the assumed complexity and makes intent for configured endpoints more explicit.

**Comment by Amwam** [comment]
I've rebased this, to add a form parameter, and removed the JSON stuff I had added pervasively (seen as that has been done now)

**Comment by Amwam** [comment]
@mrak Hi, Any chance this could be considered/looked at?

**Comment by mrak** [comment]
I've been busy, but I'll take a look later this week. Sorry that this feel through the cracks!

File: test/endpoints.js [hunk 1]
```diff
@@ -405,6 +405,52 @@ describe('Endpoints', function () {
         });
       });
 
+      describe('post versus from', function () {
```
Comment (mrak) [inline test/endpoints.js]: I'd also add a test for the negative condition: where `form` was given but does not match the form that was posted.

Similarly, where a `form` was given and it matches a subset of the post body. i.e., everything configured under `form` is present, but there were _extra_ things present in the posted form. That should still be a match.

File: src/models/endpoint.js [hunk 2]
```diff
@@ -48,6 +48,9 @@ Endpoint.prototype.matches = function (request) {
     } catch (e) {
       return null;
     }
+  } else if (this.request.form && request.post) {
+    matches.post = compareHashMaps(this.request.form, q.decode(request.post));
```
Comment (mrak) [inline src/models/endpoint.js]: Looks good.

**Comment by Amwam** [comment]
@mrak Thanks for the review, I've added the extra tests as requested, and merged in master too.
